### ISSUE TITLE
how did you generate those data isbi_train_volume.h5 and isbi_test_volume.h5

### ISSUE THREAD

**Issue by Ltieregenius** [hci-unihd/mutex-watershed#3]
Thanks for your sharing. 
I tried to open isbi_train_volume.h5 and it contains ['affinities', 'labels', 'raw'], also the f['labels'] contains 'gt_segmentation' and 'membrabes'. Are those your pre-trained results or? Do the results seem that only the mutex-watershed cannot get good segmentation results?

**Comment by constantinpape** [comment]
> Are those your pre-trained results or?

 `labels/membranes`is the membrane labeling provided by the challenge and `labels/gt_segmentation` is a 3d segmentation derived from `labels/membranes`. This is a necessary pre-processsing step in order to train the affinity network. `affinities` is the prediction of the CNN.


> Do the results seem that only the mutex-watershed cannot get good segmentation results?

Sorry, I don't understand what you mean by this question.

**Comment by Ltieregenius** [comment]
Sorry to bother you again.
Actually three days ago I followed your steps 
<img width="909" alt="image" src="https://user-images.githubusercontent.com/41847353/61168441-a646a780-a591-11e9-8e3d-dec0cb2737f8.png">
and choose the 'mws' and 'thresh' algorithms to run and it works the results as:
<img width="1089" alt="image" src="https://user-images.githubusercontent.com/41847353/61168484-3dabfa80-a592-11e9-8b55-f389f25a6ac3.png">
As we can see that the width of the cell membrane's boundary is really clear and seems to be a good segmentation result. However, when I compared these results to the ground truth:
<img width="575" alt="image" src="https://user-images.githubusercontent.com/41847353/61168525-cb87e580-a592-11e9-88c7-065df2f65b95.png">
There are still obvious differences between my output results and labels. Then I use Fiji to calculate the v-rand and the v-info:
<img width="731" alt="image" src="https://user-images.githubusercontent.com/41847353/61168553-54068600-a593-11e9-9075-86aa24ffa44a.png">
According to the evaluation results, my question is that:
(1) Do i need to do the 'dilution' to imporve the width of the boundary or do some pre-processing?
(2) could you plz point out my mistakes when i run your code like i can fix some parameters to imporve my results.

**Comment by constantinpape** [comment]
> There are still obvious differences between my output results and labels.

Of course, the algorithm will not 100 % reproduce the ground-truth. That cannot be expected. Plaese note that there are also a lot of ambiguous places in these segmentations.


> According to the evaluation results, my question is that:
> (1) Do i need to do the 'dilution' to imporve the width of the boundary or do some pre-processing?
> (2) could you plz point out my mistakes when i run your code like i can fix some parameters to imporve my results.

Yes, `dilation` might improve the results a bit, but there seems to be something more fundementally wrong with the evaluation you are running. The segmentation looks much better then the numbers you report.

As a sanity check, I would evaluate groundtruth against groundtruth and make sure that this yields a perfect score. Alternatively, use some other evaluation code, e.g.
https://github.com/cremi/cremi_python/tree/master/cremi/evaluation.
